src/Catalog.API/AIOptions.cs

```diff
@@ -9,7 +9,10 @@ public class AIOptions
 public class OpenAIOptions
 {
     /// <summary>OpenAI API key for accessing embedding LLM.</summary>
-    public string APIKey { get; set; }
+    public string ApiKey { get; set; }
+
+    /// <summary>Optional endpoint for which OpenAI API to access.</summary>
+    public string Endpoint { get; set; }
```

Comment:
I'd previously removed this as an option in order to keep eShop agnostic to Azure. I'm fine if we want to bring it back, but OpenAI won't use this; it's specific to Azure OpenAI.

Reply:
Is making it optional good enough? I left it out of the appsetting.json comment so people didn't feel like they were required to supply it.

Also - is there any documentation on how to enable the AI pieces in eshop? I couldn't find any by simple searching.

Reply:
> Also - is there any documentation on how to enable the AI pieces in eshop? I couldn't find any by simple searching.

I didn't write any, so probably not?

> Is making it optional good enough?

@adityamandaleeka?

Reply:
Just some background on why I did this - I already have an Azure account, and so it was easier for me to use Azure OpenAI than to sign up for an OpenAI account. I figured I'd save the next person this trouble in the future.

Are there other deployments of OpenAI (ex. AWS) that this Endpoint would be useful for?

Reply:
> Are there other deployments of OpenAI (ex. AWS) that this Endpoint would be useful for?

Possibly? I've seen some projects that try to emulate the OpenAI REST surface area in order to let you host a model locally but pretend to be OpenAI, in which case it's possible this would let you use that, e.g. by setting the endpoint to the localhost url where you're hosting the server, e.g.
https://github.com/mudler/LocalAI
https://github.com/THUDM/ChatGLM3
etc.

I've not tried, though.

Reply:
> I already have an Azure account, and so it was easier for me to use Azure OpenAI than to sign up for an OpenAI account

Me too - thanks!

BTW if we think plugging in completely different inference backends (e.g., Llama 2 running locally or whatever) then it would be nicer to have a C# interface people can implement rather than making them use the OpenAI REST APIs. I suppose Semantic Kernel is already meant to be an abstraction over different inference mechanisms but I don't know whether we'd need to add further abstraction.

Reply:
>  I suppose Semantic Kernel is already meant to be an abstraction over different inference mechanisms

Yes. And it's a priority for us to ensure all the right "connectors" are available for SK to make it easy to plug in different models, running both locally and remotely.